packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -4376,4 +4377,176 @@ describe("Teal Opcodes", function () {
       );
     });
   });
+
+  describe("Tealv4: Additional mathematical opcodes", () => {
+    const stack = new Stack<StackElem>();
+
+    it("divmodw", () => {
+      stack.push(0n);
+      stack.push(500n);
+      stack.push(0n);
+      stack.push(10n);
+
+      const op = new DivModw([], 1);
+      op.execute(stack);
+
+      assert.equal(stack.pop(), 0n);
+      assert.equal(stack.pop(), 0n);
+      assert.equal(stack.pop(), 50n);
+      assert.equal(stack.pop(), 0n);
+
+      stack.push(MAX_UINT64);
+      stack.push(MAX_UINT64);
+      stack.push(0n);
+      stack.push(1n);
+
+      op.execute(stack);
+
+      assert.equal(stack.pop(), 0n);
+      assert.equal(stack.pop(), 0n);
+      assert.equal(stack.pop(), MAX_UINT64);
+      assert.equal(stack.pop(), MAX_UINT64);
+
+      stack.push(MAX_UINT64);
+      stack.push(5n);
+      stack.push(MAX_UINT64);
+      stack.push(0n);
+
+      op.execute(stack);
+
+      assert.equal(stack.pop(), 5n);
+      assert.equal(stack.pop(), 0n);
+      assert.equal(stack.pop(), 1n);
+      assert.equal(stack.pop(), 0n);
+    });
+
+    it("exp", () => {
+      stack.push(2n);
+      stack.push(5n);
+      const op = new Exp([], 1);
+
+      op.execute(stack);
+      assert.equal(stack.pop(), 32n);
+
+      stack.push(5n);
+      stack.push(0n);
+
+      op.execute(stack);
+      assert.equal(stack.pop(), 1n);
+
+      stack.push(0n);
+      stack.push(1n);
+
+      op.execute(stack);
+      assert.equal(stack.pop(), 0n);
+
+      stack.push(0n);
+      stack.push(0n);
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.EXP_ERROR
+      );
+
+      stack.push(2n);
+      stack.push(66n);
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.UINT64_OVERFLOW
+      );
+    });
+
+    it("expw", () => {
+      stack.push(2n);
+      stack.push(66n);
+      const op = new Expw([], 1);
+
+      op.execute(stack);
+      const res = 2n ** 66n;
+      const low = res & MAX_UINT64;
+      const high = res >> BigInt('64');
+
+      assert.equal(stack.pop(), low);
+      assert.equal(stack.pop(), high);
+
+      stack.push(5n);
+      stack.push(0n);
+
+      op.execute(stack);
+      assert.equal(stack.pop(), 1n);
+
+      stack.push(0n);
+      stack.push(0n);
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.EXP_ERROR
+      );
+
+      stack.push(2n);
+      stack.push(128n);
```

Comment:
Same, could you please add a test where we construct a 128bit (64 + 64) output here?

Reply:
it is present at 4994 where 2^66 is a (64 + 64) output